### PR TITLE
wateranalysis: fix for empty selections and PEP8

### PR DESCRIPTION
**Pull request by orbeckst**
This is @alejob 's PR #1322 after rebase against develop and history-condensing

Changes made in this Pull Request:
 - fix wateranalysis for the case when selections are empty
 - PEP 8 formatting

This PR supersedes #1322 and closes #1322.

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - n/a Issue raised/referenced?

### CODE REVIEW

**Comment by orbeckst** [comment]
Note: This needs to be *merged* (don't do anything with rebase because otherwise the author information will get messed up).

**Comment by orbeckst** [comment]
@alejob I rewrote the history of your PR to format it as two nice separate commits:

1. fix of the bug together with test case
2. PEP 8 formatting

Please check that everything looks ok.

I did minor orthography fixes where I found them. I also reverted a few PEP8 changes in the specific cases of division and comma, e.g.
```python
val0 / n, val1 / n
```
to
```
val0/n, val1/n
```
roughly following [PEP: whitespace: Other Recommendations](https://www.python.org/dev/peps/pep-0008/#id28)... and simple aesthetics.

**Comment by orbeckst** [comment]
Sorry, had to fix something, so rebased and force-pushed.

File: package/MDAnalysis/analysis/waterdynamics.py [hunk 1]
```diff
@@ -818,33 +836,39 @@ def _sameMolecTandDT(self,selection,t0d,tf):
...
+                           interval=10, verbose=True)
         for ts in universe.trajectory:
             selection.append(universe.select_atoms(selection_str))
-            print(ts.frame)
+            pm.echo(ts.frame)
         return selection
 
-    # Second Legendre polynomial
-    lg2 = lambda self,x : (3*x*x - 1)/2
+    @staticmethod
```
Comment (alejob) [inline package/MDAnalysis/analysis/waterdynamics.py]: It makes sense.

Comment (alejob) [inline package/MDAnalysis/analysis/waterdynamics.py]: All seems good ;)

**Comment by orbeckst** [comment]
Spurious travis failure, so I restarted the job https://travis-ci.org/MDAnalysis/mdanalysis/jobs/226689879

**Comment by orbeckst** [comment]
@alejob , once travis completes successfully, your or I can merge the PR.

**Comment by alejob** [comment]
I can't merge the PR, I'm not allowed.

**Comment by orbeckst** [comment]
We need another review from an @MDAnalysis/coredevs .

> At least one approved review is required by reviewers with write access.

(I can't merge it either.)

**Comment by orbeckst** [comment]
I manually rebased against develop and force-pushed.

**Pull request merged**